staging/src/k8s.io/legacy-cloud-providers/azure/azure_wrap.go

```diff
@@ -101,7 +102,13 @@ func (az *Cloud) getVirtualMachine(nodeName types.NodeName, crt cacheReadType) (
 		return vm, cloudprovider.InstanceNotFound
 	}
 
-	return *(cachedVM.(*compute.VirtualMachine)), nil
+	virtualMachines := cachedVM.(*sync.Map)
+	if vm, ok := virtualMachines.Load(vmName); ok {
+		result := vm.(*vmEntry)
+		return *(result.vm), nil
+	}
+
+	return compute.VirtualMachine{}, nil
```

Comment:
VM not found in the cache usually means a new Node is just joining the cluster and the cache hasn't refreshed yet. So let's refresh the cache and try to get it again here.

Refer how VMSS cache hanles it https://github.com/kubernetes/kubernetes/blob/50437b4c5da29baba1f8aeef201c2b7736cd42d2/staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss.go#L94-L129.

Reply:
totally make sense. I made change to refresh vm cache.